### PR TITLE
QUIC: Fix some case in SendStreamState transition

### PR DESCRIPTION
**Pull request by scw00**
I'm really confusing with the initial and invalid state. How does it work?

### CODE REVIEW

**Comment by scw00** [comment]
```
The “Ready” state represents a newly created stream that is able to accept data from the application. 
```

File: iocore/net/quic/QUICStreamState.cc [hunk 1]
```diff
@@ -250,31 +250,26 @@ bool
...
-    }
-    break;
   case State::Send:
     if (type == QUICFrameType::STREAM || type == QUICFrameType::STREAM_BLOCKED || type == QUICFrameType::RST_STREAM) {
       return true;
     }
     break;
+  /*A sender MUST NOT send STREAM or STREAM_DATA_BLOCKED after sending a RESET_STREAM; that is, in the terminal states and in the
+   * “Reset Sent” state.*/
+  case State::ResetSent:
```
Comment (maskit) [inline iocore/net/quic/QUICStreamState.cc]: Does this mean we can RST_STREAM multiple times? I'm not sure whether we can do that.

File: iocore/net/quic/QUICStreamState.cc [hunk 2]
```diff
@@ -292,15 +287,26 @@ QUICSendStreamState::is_allowed_to_receive(const QUICFrame &frame) const
 bool
 QUICSendStreamState::is_allowed_to_receive(QUICFrameType type) const
 {
+  // An endpoint that receives a RESET_STREAM frame for a send-only stream MUST terminate the connection with error
+  // PROTOCOL_VIOLATION. 
+  // PS: RST_STREAM should be processed before Stream transition.
```
Comment (maskit) [inline iocore/net/quic/QUICStreamState.cc]: I don't think we should assume this. We cannot send PROTOCOL_VIOLATION (from the caller) if we assert here.

**Comment by maskit** [comment]
Please add test cases.

**Comment by scw00** [comment]
Close temporarily and will push again if we finished the send buffer.

**Pull request closed without merging**